Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -553,6 +553,130 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Push and Pull Attachments SG", "[.
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+        Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+        collectionPreamble(collectionSpecs, "sguser", "password", kC4LogNone);
+    std::array<unordered_map<alloc_slice, unsigned>, collectionCount> docIDs;
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+    std::array<string, collectionCount> collNames = {"rose"};
+
+    for (size_t i = 0; i < collectionCount; ++i) {
+        createFleeceRev(collections[i], slice(idPrefix+collNames[i]), kRev1ID, "{}"_sl);
+        createFleeceRev(collections[i], slice(idPrefix+collNames[i]), revOrVersID("2-12121212", "1@cafe"),
+                        "{\"db\":\"remote\"}"_sl);
+        docIDs[i] = getDocIDs(collections[i]);
+        replCollections[i] = C4ReplicationCollection{collectionSpecs[i], kC4OneShot, kC4Disabled};
+    }
+
+    // Send the docs to remote
+    C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+    replicate(paramsSetter);
+    verifyDocs(collectionSpecs, docIDs, true);
+    
+    deleteAndRecreateDB();
+    for (size_t i = 0; i < collectionCount; ++i) {
+        collections[i] = c4db_createCollection(db, collectionSpecs[i], ERROR_INFO());
+        createFleeceRev(collections[i], slice(idPrefix+collNames[i]), kRev1ID, "{}"_sl);
+        createFleeceRev(collections[i], slice(idPrefix+collNames[i]), revOrVersID("2-13131313", "1@babe"),
+                        "{\"db\":\"local\"}"_sl);
+        replCollections[i] = C4ReplicationCollection{collectionSpecs[i], kC4Disabled, kC4OneShot};
+    }
+    std::array<AllocedDict, collectionCount> docIDsDicts;
+    paramsSetter = [&replCollections, &docIDs, &docIDsDicts,
+                     collectionCount](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+        for (size_t i = 0; i < collectionCount; ++i) {
+            fleece::Encoder enc;
+            enc.beginArray();
+            for (const auto& d : docIDs[i]) {
+                enc.writeString(d.first);
+            }
+            enc.endArray();
+            Doc doc {enc.finish()};
+            docIDsDicts[i] =
+                repl::Options::updateProperties(
+                    AllocedDict(c4Params.collections[i].optionsDictFleece),
+                    kC4ReplicatorOptionDocIDs,
+                    doc.root());
+            c4Params.collections[i].optionsDictFleece = docIDsDicts[i].data();
+        }
+    };
```

Comment:
If this needs to be done on every test, I think we should have a utility function for this.

Reply:
I can certainly do it, but I feel every test case should have its parameter setting. The first two lines are absolutely common, but the rest are per-test.

Reply:
The test can setup the param but have a function that takes some params to setup docIDs? It is really adding some complexity to the test. Jim is working on some idea in this area, hope that can be applied here as well.

Reply:
yes, docIDs are used to work around the issue with flushing before every test. Let's see if we want to abandon this approach altogether.